sdk/core/azure_core_test/src/proxy/policy.rs

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+use async_trait::async_trait;
+use azure_core::{
+    headers::{AsHeaders, HeaderName, HeaderValue},
+    test::TestMode,
+    Context, Policy, PolicyResult, Request,
+};
+use std::{
+    convert::Infallible,
+    sync::{Arc, RwLock},
+};
+use tracing::{debug_span, Instrument};
+
+use crate::Skip;
+
+#[derive(Debug, Default)]
+pub struct RecordingPolicy {
+    pub test_mode: TestMode,
+    pub options: RwLock<RecordingOptions>,
```

Comment:
@analogrelay wondering if you have any better ideas. Basically, I want the policy (in concept - not necessarily this implementation) to send headers as needed. And while there *should* be a 1:1 correlation between `Recording` and `RecordingPolicy`, separate threads are certainly a possibility and this, I believe, upholds the intent of keeping a vector of `Arc<dyn Policy>` as the pipeline: to share policies as appropriate.

Reply:
I had contemplated that, but test code would have to remember to inherit the `Context` for every call into a client, which is error prone. Then there are the implicit calls e.g., for credentials. A macro *could* certainly do that but would be much more expensive to develop. Perhaps a goal further down the road.

Reply:
Thinking about this more, I guess I could see a case where callers could do this as needed like so:

```rust
let ctx = Context::with_value(SkipRecording::RequestBody);
let options = ClientMethodOptions { context: ctx, ..Default::default() };
client.foo(..., Some(ClientFooOptions { method_options: options.clone(), ..Default::default() })).await?;
client.bar(..., None).await?;
```

@LarryOsterman or @RickWinter, thoughts?

Reply:
My biggest issue here is that I *think* that means we end up with two different ways of doing the same thing, which can be confusing. OTOH, the only people being confused would be SDK client developers.

And if we remove skip from the RecordOptions (to make it so that there's only one way of doing that), then we have some recording options in the Context and some others in RecordOptions, which is just *weird*.

Reply:
I was thinking of supporting skipping only one way, but you make a good point about having other options on the `recording` object.